src/common/utils.h

```diff
@@ -663,6 +663,18 @@ constexpr size_t MaxIntegerValue<mshadow::half::half_t>() {
   return size_t(2) << 10;
 }
 
+MSHADOW_XINLINE int ilog2ul(size_t a) {
```

Comment:
Looks good. Please also refactor other places using the same utility as we discussed.

Reply:
Done.